packages/devops/scripts/lambda/actions/redeploy_tupaia_database.py

```diff
@@ -0,0 +1,48 @@
+# Drops and recreates a tupaia database instance. Useful for updating to latest snapshot data
+#
+# Example configs
+#
+# 1. Drop and redeploy a tupaia database instance
+# {
+#   "Action": "redeploy_tupaia_database",
+#   "DeploymentName": "dev"
+# }
+#
+
+import asyncio
+
+from helpers.creation import create_db_instance_from_snapshot
+from helpers.teardown import teardown_db_instance
+from helpers.rds import get_db_instance, wait_for_db_instance, set_db_instance_master_password
+from helpers.secrets import get_db_master_password
+
+loop = asyncio.get_event_loop()
+
+def redeploy_tupaia_database(event):
+    # validate input config
+    if 'DeploymentName' not in event:
+        raise Exception('You must include the key "DeploymentName" in the lambda config, e.g. "dev".')
+    deployment_name = event['DeploymentName']
+
+    db_id = 'tupaia-' + deployment_name
+    db_instance = get_db_instance(db_id)
+    # get manual input parameters, or default for any not provided
+    db_instance_type = db_instance['DBInstanceClass']
+    security_group_id = db_instance['VpcSecurityGroups'][0]['VpcSecurityGroupId']
+    clone_db_from = next(filter(lambda item: item['Key'] == 'ClonedFrom', db_instance['TagList']))['Value']
+
+    # rename then delete existing db
+    teardown_db_instance(deployment_name, 'tupaia')
+    loop.run_until_complete(wait_for_db_instance(db_id, 'deleted'))
```

Comment:
This does create a bit of a database downtime, but given it happens overnight and our feature instance should be offline then anyway it ought to be fine. The alternative is to spin up the new one under a new name and once it's up drop the old one and rename, but the timing for that was proving tricky within the 15 min timeout window for lambdas.